mousestyles/classification/clustering.py

```diff
@@ -0,0 +1,291 @@
+from matplotlib import pyplot as plt
+from scipy.cluster.hierarchy import dendrogram, linkage
+from sklearn.cluster import AgglomerativeClustering, KMeans
+from sklearn import metrics
+import numpy as np
+from scipy.cluster.hierarchy import cophenet
+from scipy.spatial.distance import pdist
+from mousestyles.data.utils import day_to_mouse_average
+
+
+# prep data functions
+def prep_data(mouse_data, std=True, rescale=True):
+    """
+    Returns a ndarray data to be used in clustering algorithms:
+        column 0 : strain,
+        column 1: mouse,
+        other columns corresponding to feature avg/std of a mouse over 16 days
+            that may or may not be rescaled to the same unit as specified
+
+    Parameters
+    ----------
+    mouse_data: a 21131 * 13 pandas DataFrame,
+        column 0 : strain,
+        column 1: mouse,
+        column 2: day,
+        column 3: hour,
```

Comment:
So what if this kind of data frame is not inputted. For example, what if the mouseday data is inputted and there is only 3 column before actual features instead of 4.
Can we either do some check before starting the main algorithm, or be more flexible about the input data frame?